templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -751,18 +749,30 @@ contents:
     # Setup an exit trap to rollback on error
     handle_exit() {
       e=$?
-      [ $e -eq 0 ] && print_state && exit 0
+      tdir=$(mktemp -u -d -t "configure-ovs-$(date +%Y-%m-%d-%H-%M-%S)-XXXXXXXXXX")
+      
+      if [ $e -eq 0 ]; then
+        print_state
+        # remove previous troubleshooting information
+        rm -rf "$(dirname "$tdir")/configure-ovs-*"
```

Comment:
`dirname`  could be done with bash param expansion as `${tdir%/*}`

Reply:
Ack, since it was already being used elsewhere and should pose no functional issue, we can change those next chance we have.

Reply:
I'd actually prefer not to. Bash magic is all well and good, but it's extremely unreadable and difficult to even search on if you don't know what a particular expression does. Googling dirname is much easier than googling %/*. There are cases where the alternative is unwieldy, but I don't think a single call to dirname is one of them.

Reply:
It's to avoid having to worry about word splitting and quoting issues with subshells.

Either way seems to work with spaces and newlines in `$TMPDIR`.

Reply:
Also obligatory  `rm -rf --`  suggestion.

Reply:
Interesting behavior with `dirname --` and `TMPDIR`

```shell
+ export 'TMPDIR=--no-preserve-root '
+ TMPDIR='--no-preserve-root '
+++ date +%Y-%m-%d-%H-%M-%S
++ mktemp -u -d -t configure-ovs-2023-03-03-13-20-45-XXXXXXXXXX
+ tdir='--no-preserve-root /configure-ovs-2023-03-03-13-20-45-dtLNTZdIEB'
++ dirname '--no-preserve-root /configure-ovs-2023-03-03-13-20-45-dtLNTZdIEB'
dirname: unrecognized option '--no-preserve-root /configure-ovs-2023-03-03-13-20-45-dtLNTZdIEB'
Try 'dirname --help' for more information.
+ echo rm -rf '/configure-ovs-*'
rm -rf /configure-ovs-*
++ dirname -- '--no-preserve-root /configure-ovs-2023-03-03-13-20-45-dtLNTZdIEB'
rm -rf --no-preserve-root /configure-ovs-*
with ${tdir%/*}
rm -rf --no-preserve-root /configure-ovs-*
+ echo rm -rf '--no-preserve-root /configure-ovs-*'
+ echo 'with ${tdir%/*}'
+ echo rm -rf '--no-preserve-root /configure-ovs-*'
```